gitflow_toolbox/common/gitlab.py

```diff
@@ -0,0 +1,55 @@
+import os
+
+import gitlab
+from gitlab.v4.objects.projects import Project
+
+from gitflow_toolbox.common.get_env import get_env
+from gitflow_toolbox.common.singleton import Singleton
+
+
+class CurrentGitlab(gitlab.Gitlab, metaclass=Singleton):
+
+    current_project_memo = None
+
+    def __init__(self):
+        # If running in a gitlab CI job
+        if os.getenv("GITLAB_CI"):
+            # Use provided token, if not available, use default token from job
+            private_token = get_env("GITLAB_PRIVATE_TOKEN", "CI_JOB_TOKEN")
+            super().__init__(get_env("CI_SERVER_URL"), private_token)
+        else:
+            super().__init__(get_env("GITLAB_URL"), get_env("GITLAB_PRIVATE_TOKEN"))
+
+    def current_project(self) -> Project:
+        """Get current gitlab sdk project
+
+        Returns:
+            gitlab.Gitlab: current gitlab project instance (or 'source', 'from' gitlab project)
+        """
+        if not self.current_project_memo:
```

Comment:
Use a private attribute of the class (to initialise in __init__() )

`self.__current_project`